source/reflect.h2

```diff
@@ -1167,40 +1167,98 @@ basic_enum: (
     t.add_member( "    operator=         : (out this, that) == { }");
     t.add_member( "    operator<=>       : (this, that) -> std::strong_ordering;");
 
-    //  Provide a 'to_string' function to print enumerator name(s)
-    (copy to_string: std::string = "    to_string: (this) -> std::string = { \n")
+    //  Provide 'to_string' and 'to_code' functions to print enumerator
+    //  name(s) as human-readable strings or as code expressions
+    (copy to_string_impl: std::string = "    to_string_impl: (this, prefix: std::string_view")
     {
         if bitwise {
-            to_string += "    _ret   : std::string = \"(\";\n"
-                         "    _comma : std::string = ();\n"
-                         "    if this == none { return \"(none)\"; }\n";
+            to_string_impl += ", separator: std::string_view ) -> std::string = { \n"
+                              "    ret : std::string = \"(\";\n"
+                              "    sep : std::string = ();\n"
+                              "    if this == none { return \"(none)\"; }\n";
+        }
+        else {
+            to_string_impl += ") -> std::string = { \n";
         }
 
         for  enumerators
         do   (e) {
             if e.name != "_" {  // ignore unnamed values
                 if bitwise {
                     if e.name != "none" {
-                        to_string += "    if (this & (e.name)$) == (e.name)$ { "
-                                             "_ret += _comma + \"(e.name)$\"; _comma = \", \"; "
-                                         "}\n";
+                        to_string_impl += "    if (this & (e.name)$) == (e.name)$ { "
+                                                  "ret += sep + cpp2::to_string(prefix) + \"(e.name)$\"; sep = separator; "
```

Comment:
This results in code like this:

```
if (((*this) & case_insensitive) == case_insensitive) {ret += sep + cpp2::to_string(prefix) + "case_insensitive";sep = separator;}
if (((*this) & multiple_lines) == multiple_lines) {ret += sep + cpp2::to_string(prefix) + "multiple_lines";sep = separator;}
if (((*this) & single_line) == single_line) {ret += sep + cpp2::to_string(prefix) + "single_line";sep = separator;}
```

This calls `sep + cpp2::to_string(prefix)` for every matched bit.

Could the function start something like this:

```
    auto first = cpp2::to_string(prefix);
    auto next = separator + first;
    auto cur = &first;
```
and then for each bit
```
    if (((*this) & case_insensitive) == case_insensitive) {ret += cur + "case_insensitive"; cur = &next;}
```